mixer is static, classes changed

vpligin1127 opened the pull request:
Переписаны классы Car1  и Car2, метод mixer() сделан static

vpligin1127 committed:
mixer is static, classes changed

Переписаны классы Car1  и Car2, метод mixer() сделан static

eevtukhova commented on .idea/vcs.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
папки .idea, out, target - коммитить не надо, рекомендую почитать про файл gitignore - чтобы они никогда автоматически не коммитились https://git-scm.com/docs/gitignore

eevtukhova commented on src/solution/Car1.java:
```diff
@@ -6,24 +6,24 @@
+    private  static int maxSpeed = 380;
+    private  static int minSpeed = 0;
+    private  static int engineNumber = r.nextInt();
     private  static String driverName = "Trololoshkin";
```
предзаполнять поля в pojo классах не стоит (разве только в случае с дефолтными значениями полей, но здесь это не требуется). заполнять нужно через сеттеры после вызова new

eevtukhova commented on src/solution/Car2.java:
```diff
@@ -6,50 +6,50 @@
 
-    static String getBrand(Car2 car2) {
-        return car2.brand;
+    static String getBrand() {
```
геттеры-сеттеры не должны быть статиками

eevtukhova commented on src/solution/Car2.java:
```diff
@@ -6,50 +6,50 @@
+    private static int maxSpeed = 80;
+    private static int minSpeed = 0;
+    private static int engineNumber = r.nextInt();
+    private static String driverName = "mrSpeedy";
```
и поля тоже не должны быть статичными

eevtukhova commented on src/solution/CarHacker.java:
```diff
@@ -2,31 +2,31 @@
-        String tmpBrand = car1.getBrand();
-        Car1.setBrand(Car2.getBrand(car2));
-        car2.setBrand(tmpBrand);
+        String tmpBrand = Car1.getBrand();
```
опыть же - у каких машин меняются поля? должны поменять у тех инстансов, которые созданы в main:Car1 car1 = new Car1();
                Car2 car2 = new Car2();

vpligin1127 committed:
comments

исправлено по замечаниям
создаются инстансы в main, заполняются поля через сеттеры
в mixer() передаются созданные инстансы

eevtukhova commented on .idea/workspace.xml:
```diff
@@ -17,55 +16,55 @@
     <option name="LAST_RESOLUTION" value="IGNORE" />
```
повторюсь - папки .idea, out, target - коммитить НЕ НАДО, рекомендую почитать про файл gitignore - чтобы они никогда автоматически не коммитились https://git-scm.com/docs/gitignore

eevtukhova commented on src/solution/Car2.java:
```diff
@@ -1,55 +1,60 @@
-public class Car2 {
+//import java.util.Random;
 
+public class Car2 extends Car1{
```
можно в этом случае вообще не экстендить класс, а просто создавать два экземпляра одного класса Car1:
Car1 car1 = new Car1();
Car1 car2 = new Car1();

vpligin1127 committed:
gitignore

gitignore added

vpligin1127 committed:
gitignore

changed

vpligin1127 committed:
Update .gitignore

vpligin1127 committed:
ignored deleted

ignoring folders

vpligin1127 committed:
Merge branch 'changed_classes' of https://github.com/vpligin1127/method_levels into changed_classes
